### DISCUSSION TITLE
HTTP Toolkit Android integration for mitmproxy

### DISCUSSION THREAD

**Discussion by hacker1024** [mitmproxy/mitmproxy#6209] [Intercept Everything!]
I have made an addon that lets you use the [HTTP Toolkit] Android app with mitmproxy.
This makes connecting an Android device very easy.

https://github.com/hacker1024/mitmproxy_httptoolkit_android

[HTTP Toolkit]: https://httptoolkit.com

**Comment by mhils** [comment]
This is interesting, thanks! I have not played around with the HTTP Toolkit Android app yet. Does this require a rooted device? Have you by chance used our WireGuard mode, and could make a comparison?

**Reply by hacker1024** [reply]
> Does this require a rooted device?

It does not, though most Android apps will not accept TLS certificates that are not installed in the system, which requires root access.

> Have you by chance used our WireGuard mode, and could make a comparison?

I have not used this mode. The main benefits of the HTTP Toolkit app are guided certificate installation and per-app interception toggles, which may be hard to do with WireGuard.

**Reply by mhils** [reply]
WireGuard does per-app interception on Android for us, but yes, absolutely no cert install. :)
